customize_image.sh

```diff
@@ -255,24 +292,27 @@ if [ -e "$IMAGEDIR/etc/init.d/apache2" ] ; then
 		"$IMAGEDIR/etc/apache2/apache2.conf"
 fi
 
-# install nodejs and nodered with an install script and revpi-nodes from npm repository
-NODEREDSCRIPT="/tmp/update-nodejs-and-nodered.sh"
-/usr/bin/curl -sL \
-	https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered\
-	--output "$IMAGEDIR/$NODEREDSCRIPT"
-chmod 755 "$IMAGEDIR/$NODEREDSCRIPT"
-chroot "$IMAGEDIR" /usr/bin/sudo -u pi $NODEREDSCRIPT --confirm-install --confirm-pi
-rm "$IMAGEDIR/$NODEREDSCRIPT"
-chroot "$IMAGEDIR" /usr/bin/sudo -u pi /usr/bin/npm install --prefix /home/pi/.node-red node-red-contrib-revpi-nodes
-
+if [ "$MINIMG" != "1" ]; then
+	# install nodejs and nodered with an install script and revpi-nodes from npm repository
+	NODEREDSCRIPT="/tmp/update-nodejs-and-nodered.sh"
+	/usr/bin/curl -sL \
+		https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered\
+		--output "$IMAGEDIR/$NODEREDSCRIPT"
+	chmod 755 "$IMAGEDIR/$NODEREDSCRIPT"
+	chroot "$IMAGEDIR" /usr/bin/sudo -u pi $NODEREDSCRIPT --confirm-install --confirm-pi
+	rm "$IMAGEDIR/$NODEREDSCRIPT"
+	chroot "$IMAGEDIR" /usr/bin/sudo -u pi /usr/bin/npm install --prefix /home/pi/.node-red node-red-contrib-revpi-nodes
+fi
 # enable ssh daemon by default, disable swap, disable bluetooth on mini-uart
 chroot "$IMAGEDIR" systemctl enable ssh
 chroot "$IMAGEDIR" systemctl disable dphys-swapfile
 chroot "$IMAGEDIR" systemctl disable hciuart
 
 # disable 3rd party software
-chroot "$IMAGEDIR" systemctl disable logiclab
-chroot "$IMAGEDIR" systemctl disable nodered
+if [ "$MINIMG" != "1" ]; then
+	chroot "$IMAGEDIR" systemctl disable logiclab
+	chroot "$IMAGEDIR" systemctl disable nodered
+fi
 chroot "$IMAGEDIR" systemctl disable noderedrevpinodes-server
```

Comment:
Would it make sense to group all additions for a "full" image into one if instead of cluttering it all over the file? Or are I'm missing some dependencies?

```
chroot "$IMAGEDIR" apt-get -y install `egrep -v '^#' "$BAKERYDIR/min-debs-to-download"`
dpkg --root "$IMAGEDIR" --force-depends --purge rpd-wallpaper
chroot "$IMAGEDIR" apt-get -y install revpi-wallpaper
chroot "$IMAGEDIR" apt-get update

if [ "$MINIMG" != "1" ]; then
	chroot "$IMAGEDIR" apt-get -y install `egrep -v '^#' "$BAKERYDIR/debs-to-download"`
	chroot "$IMAGEDIR" apt-get -y install teamviewer-revpi
	
	# install nodejs and nodered with an install script and revpi-nodes from npm repository
	NODEREDSCRIPT="/tmp/update-nodejs-and-nodered.sh"
	/usr/bin/curl -sL \
		https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered\
		--output "$IMAGEDIR/$NODEREDSCRIPT"
	chmod 755 "$IMAGEDIR/$NODEREDSCRIPT"
	chroot "$IMAGEDIR" /usr/bin/sudo -u pi $NODEREDSCRIPT --confirm-install --confirm-pi
	rm "$IMAGEDIR/$NODEREDSCRIPT"
	chroot "$IMAGEDIR" /usr/bin/sudo -u pi /usr/bin/npm install --prefix /home/pi/.node-red node-red-contrib-revpi-nodes

	chroot "$IMAGEDIR" systemctl disable logiclab
	chroot "$IMAGEDIR" systemctl disable nodered
fi

# enable ssh daemon by default, disable swap, disable bluetooth on mini-uart
chroot "$IMAGEDIR" systemctl enable ssh
chroot "$IMAGEDIR" systemctl disable dphys-swapfile
chroot "$IMAGEDIR" systemctl disable hciuart
```

Reply:
yes, as I remember, there are some dependencies between them.
Would think they can be removed somehow, however, it seems is another topic.

Reply:
@ferdrich-kunbus I like your idea of moving most of the additional parts into one group, but would suggest to address this in a separate PR